APIImpl/PluginManager.lua

```diff
@@ -42,18 +42,16 @@ return
 
 	["cPluginManager:GetCurrentPlugin()"] = function (a_Simulator, a_Self)
 		local res = a_Simulator:createInstance({Type = "cPluginLua"})
-		-- Override the plugin to act as "self":
-		getmetatable(res).__index.GetLocalFolder = function(a_Self)
-			if not(a_Self == res) then
-				error("Bad cPlugin:GetLocalFolder() usage, the first parameter is not \"self\".")
-			end
-			return a_Simulator.options.pluginPath
-		end
+		-- Save the path to the plugin folder in the self variable
+		getmetatable(res).__pluginPath = a_Simulator.options.pluginPath
```

Comment:
So far the `simulatorInternal_` prefix has been used for all internal simulator variables inside the classes, so it should be used here as well.